lib/pg_database.rb

```diff
@@ -0,0 +1,39 @@
+require 'pg'
+require_relative '../models/game'
+require_relative '../serializers/game_serializer'
+
+class PGDatabase
+  def initialize(data = Game, serializer = GameSerializer.new)
+    @data = data
+    @serializer = serializer
+  end
+
+  def store(id, game_data)
+    @data.find_or_initialize_by(game_id: id).update!(game_data || {})
+  end
+
+  def load_game(game_id)
+    game = @data.find_by(game_id: game_id)
+    @serializer.serialize(game)
+  end
+
+  def key?(game_id)
+    @data.exists?(game_id: game_id)
+  end
+
+  def clear; end
+
+  def load(game_id, key)
+    game = @data.find_by(game_id: game_id)
+    @serializer.serialize(game).dig(key)
+  end
+
+  def load_all
+    game_hash = {}
+    @data.find_each do |game|
+      game_data = @serializer.serialize(game)
+      game_hash.store(game.game_id, game_data)
+    end
+    game_hash
```

Comment:
As discussed yesterday, a slightly refactor to remove the additional `game_hash` variable is to use [inject/reduce](https://apidock.com/ruby/Enumerable/inject).

```rb
@data.all.inject({}) do |result, record|
  result[record.game_id] = @serializer.serialize(record)
  result
end
```